apps/app-article/src/App.js

```diff
@@ -68,23 +72,38 @@ class App extends Component {
             this.setState({fontSize: parseFloat(localStorage.getItem("fontSize"))});
         }
 
-        this.getArticle();
-        this.getMostReadArticles();
+        //In case User want to logout, the value should be false
+        if(Cookies.get('LoginStatus') != undefined && !Cookies.get('LoginStatus')){
+            //we remove it to avoid infinite loop
+            Cookies.remove('LoginStatus');
+            //TODO
+            // we need to have logout listener here, after success we can then remove the cookie, if for exemple an error happend while             
+            //LogOut;in localstorage will keep logged but for android is considered as logged out
+
+            logout(this.onLogout);
+        }
+        if(getUrlParam().has('login')){
+            this.setState({forceLoginView: true});
+        }else {
+            this.getArticle();
+            this.getMostReadArticles();
+        }
     }
 
+    onLogout() {
+        console.log("Logged out successfully!")
+    }    
     getArticle() {
         if (JSON.parse(localStorage.getItem('cachedArticles')) != null) {
             this.setState({cachedArticles: JSON.parse(localStorage.getItem('cachedArticles'))});
         }
-
-        let urlParams = new URLSearchParams(window.location.search);
-        if (urlParams.has('uuid')) {
+        if (getUrlParam().has('uuid')) {
             if (this.checkCache(urlParams.get('uuid'))) {
                 this.fetchArticleFromCache(urlParams.get('uuid'));
             } else {
                 this.fetchArticleFromApi(urlParams.get('uuid'));
```

Comment:
where do we get urlParams from as the declaration is removed?

Reply:
@mrskug fixed now, thanks for remark :)